Library/Homebrew/utils/curl.rb

```diff
@@ -198,20 +198,40 @@ def curl_output(*args, **options)
     end
 
     # Check if a URL is protected by CloudFlare (e.g. badlion.net and jaxx.io).
+    # @param details [Hash] Response information from
+    #  `#curl_http_content_headers_and_checksum`.
+    # @return [true, false] Whether a response contains headers indicating that
+    #   the URL is protected by Cloudflare.
+    sig { params(details: T::Hash[Symbol, T.untyped]).returns(T::Boolean) }
     def url_protected_by_cloudflare?(details)
       return false if details[:headers].blank?
-      return unless [403, 503].include?(details[:status].to_i)
+      return false unless [403, 503].include?(details[:status].to_i)
 
-      details[:headers].fetch("set-cookie", nil)&.match?(/^(__cfduid|__cf_bm)=/i) &&
-        details[:headers].fetch("server", nil)&.match?(/^cloudflare/i)
+      set_cookie_header = Array(details[:headers]["set-cookie"])
+      has_cloudflare_cookie_header = set_cookie_header.compact.any? do |cookie|
+        cookie.match?(/^(__cfduid|__cf_bm)=/i)
+      end
+
+      server_header = Array(details[:headers]["server"])
+      has_cloudflare_server = server_header.compact.any? do |server|
+        server.match?(/^cloudflare/i)
+      end
```

Comment:
For what it's worth, I converted this to also use the `Array()`/`any?` approach, as `details[:headers]["server"].match?(...)` would cause Sorbet to complain that `details[:headers]["server"]` can potentially be a type that doesn't respond to `match?` (i.e., an array). Under normal circumstances, I don't believe a `server` header should appear more than once (i.e., it should be a string, if present) but we can't tell this to Sorbet.

At the end of the day, we rely on an upstream server to do things correctly and that may not always be the case, so the "better safe than sorry" approach kind of makes sense anyway.